### PR TITLE
Dockerized Chai training gym

### PR DESCRIPTION
**Pull request by RRustom**
- Created a new docker service for the Chai training gym
- The training should work, but is far from optimized. This will be the subject of upcoming PRs. This should unblock us from working in parallel to start finetuning and optimizing the training process!
- Instructions on how to get it setup can be found in `/chai_research/README.md`

### CODE REVIEW

File: chai_research/scripts/train/explorations/lmgym_files/deepspeed_configs/ds_config_small.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
```
Comment (MarkRich) [inline chai_research/scripts/train/explorations/lmgym_files/deepspeed_configs/ds_config_small.json:1]: I think we can delete this plus the parent directory right?

File: chai_research/.gitignore [hunk 2]
```diff
@@ -1,4 +1,7 @@
 data
 scripts/train/explorations/logs
 *wandb*
-*.DS_Store
\ No newline at end of file
+*.DS_Store
+*.env
+*__pycache__*
```
Comment (MarkRich) [inline chai_research/.gitignore]: Looks like this is not fulling taking effect? still some __pycache__ directories being added which we can prune.

File: chai_research/lmgym_train/train/explorations/lmgym/clm_models/callbacks/callbacks.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            pattern=os.path.join(training_args.output_dir, 'checkpoint-*', 'global_step*')
+        )
+        callbacks.append(callback)
+    return callbacks
+
+
+def _load_reward_model(name, local_rank):
+    model = AutoModelForSequenceClassification.from_pretrained(
+        name,
+        use_auth_token=True
```
Comment (MarkRich) [inline chai_research/lmgym_train/train/explorations/lmgym/clm_models/callbacks/callbacks.py:65]: Deprecated? Not a part of the code we call but I guess consistency.

**Comment by MarkRich** [comment]
Looks great; few small comments but other than that feel free to merge.

File: chai_research/scripts/train/explorations/lmgym_files/deepspeed_configs/ds_config_small.json (hunk 1 above)
Comment (RRustom) [reply chai_research/scripts/train/explorations/lmgym_files/deepspeed_configs/ds_config_small.json:1]: We're still using the `ds_config_small` in the `hh.sh`

**Pull request merged**